Rakefile

```diff
@@ -8,5 +8,13 @@ Rake::TestTask.new(:test) do |t|
 end
 
 require 'rake/extensiontask'
-Rake::ExtensionTask.new("syslog")
+Rake::ExtensionTask.new("syslog_ext") do |ext|
+  ext.ext_dir = 'ext/syslog'
+
+  # In contrast to "gem install" a "rake compile" is expecting the C-ext file even on Windows.
+  # Work around by creating a dummy so file.
+  task "#{ext.tmp_dir}/#{ext.platform}/stage/lib" do |t|
+    touch "#{ext.tmp_dir}/#{ext.platform}/#{ext.name}/#{RUBY_VERSION}/#{ext.name}.so"
```

Comment:
This workaround is necessary to succeed `rake compile` on Windows (without a `syslog_ext.so` file). Alternatives would be:

1. Create a dummy `syslog_ext.so` in the Makefile generated by extconf.rb
2. Change rake-compiler to allow `compile` task without a extension file
3. Change the CI jobs to `gem build` and `gem install` instead of `rake compile`

The workaround is also necessary when not renaming the C-ext to `syslog_ext.so`.